pkg/applicationserver/io/packages/loragls/v3/data.go

```diff
@@ -131,6 +141,95 @@ func toFloat64(f float64) *structpb.Value {
 	}
 }
 
+func toGatewayIDsStruct(gatewayIDs *ttnpb.GatewayIdentifiers) *structpb.Value {
```

Comment:
I think we should cheat a little and use something like [gob](https://pkg.go.dev/encoding/gob). We can use our local struct definitions here as `gob` messages, and only store the binary as base64 in `data`.

The reason is that external users should not interact with the recent metadata (they don't insert it anyway), and having the long field names is inefficient space wise.

This has the advantage that we don't need to write encoders and decoders by hand - we just marshal to base64 and back.